cpp/src/hash/md5_hash.cu

```diff
@@ -27,23 +27,314 @@
+      thrust::fill_n(
+        thrust::seq,
+        hash_state->buffer + hash_state->buffer_length + 1,
+        (md5_chunk_size - message_length_size - end_of_message_size - hash_state->buffer_length),
+        0x00);
+    } else {
+      thrust::fill_n(thrust::seq,
+                     hash_state->buffer + hash_state->buffer_length + 1,
+                     (md5_chunk_size - hash_state->buffer_length),
+                     0x00);
+      md5_hash_step(hash_state);
+
+      thrust::fill_n(thrust::seq, hash_state->buffer, md5_chunk_size - message_length_size, 0x00);
+    }
+
+    std::memcpy(hash_state->buffer + md5_chunk_size - message_length_size,
+                reinterpret_cast<uint8_t const*>(&full_length),
+                message_length_size);
+    md5_hash_step(hash_state);
+
+#pragma unroll
+    for (int i = 0; i < 4; ++i)
+      uint32ToLowercaseHexString(hash_state->hash_value[i], result_location + (8 * i));
+  }
+
+  template <typename T, std::enable_if_t<is_chrono<T>()>* = nullptr>
+  void __device__ operator()(column_device_view col,
+                             size_type row_index,
+                             md5_intermediate_data* hash_state) const
+  {
+    cudf_assert(false && "MD5 Unsupported chrono type column");
+  }
+
+  template <typename T, std::enable_if_t<!is_fixed_width<T>()>* = nullptr>
+  void __device__ operator()(column_device_view col,
+                             size_type row_index,
+                             md5_intermediate_data* hash_state) const
+  {
+    cudf_assert(false && "MD5 Unsupported non-fixed-width type column");
+  }
+
+  template <typename T, std::enable_if_t<is_floating_point<T>()>* = nullptr>
+  void __device__ operator()(column_device_view col,
+                             size_type row_index,
+                             md5_intermediate_data* hash_state) const
+  {
+    md5_process(normalize_nans_and_zeros_helper<T>(col.element<T>(row_index)), hash_state);
+  }
+
+  template <
+    typename T,
+    std::enable_if_t<is_fixed_width<T>() && !is_floating_point<T>() && !is_chrono<T>()>* = nullptr>
+  void CUDA_DEVICE_CALLABLE operator()(column_device_view col,
+                                       size_type row_index,
+                                       md5_intermediate_data* hash_state) const
+  {
+    md5_process(col.element<T>(row_index), hash_state);
+  }
+};
+
+template <>
+void CUDA_DEVICE_CALLABLE MD5Hash::operator()<string_view>(column_device_view col,
+                                                           size_type row_index,
+                                                           md5_intermediate_data* hash_state) const
+{
+  string_view key     = col.element<string_view>(row_index);
+  uint32_t const len  = static_cast<uint32_t>(key.size_bytes());
+  uint8_t const* data = reinterpret_cast<uint8_t const*>(key.data());
+
+  hash_state->message_length += len;
+
+  if (hash_state->buffer_length + len < 64) {
```

Comment:
Magic number is still here, Probably want to do the same. Maybe just define a namespace level constant rather than a function local one in every single operator?

Reply:
Good catch. I'm fixing this with a local variable for now -- this will be changed to use a class-level constant in a follow-up PR that uses the hashing base class I'm writing for #9215.